tests/testthat/test-lint_file.R

```diff
@@ -225,3 +225,18 @@ test_that("Deprecated positional usage of cache= works, with warning", {
   )
   expect_identical(l, lint("a = 2\n", assignment_linter(), cache = FALSE))
 })
+
+test_that("Linters throwing an error give a helpful error", {
+  tmp_file <- withr::local_tempfile(lines = "a <- 1")
+  linter <- function() Linter(function(source_expression) stop("a broken linter"))
+  # NB: Some systems/setups may use e.g. symlinked files when creating under tempfile();
+  #   we don't care much about that, so just check basename()
+  expect_error(
+    lint(tmp_file, linter()),
+    paste0("Linter 'linter' failed in .*", basename(tmp_file), ": a broken linter")
```

Comment:
`basename(...)` could need regex escaping. Use `rex::rex()` instead of `paste0()`?

Reply:
I don't think it's true because R generates temp files from alphanumeric characters, but it can't hurt to use `rex()` instead. done.